docs/source/migration.mdx

```diff
@@ -47,20 +40,157 @@ graph TB;
 
 If you are using any other Apollo Server 3 framework integration package, you can’t upgrade during the AS4 alpha release _yet_. Please help us by [building new integrations](/building-integrations) or [discussing how to maintain existing integrations](https://github.com/apollographql/apollo-server/labels/integration-collaborators) to ensure there is an Apollo Server 4 integration for your favorite framework.
 
-Apollo Server 4 also removes the built-in dependency on the `cors` and `body-parser` libraries (excluding the `startStandaloneServer` function).
+There are a few other high-level changes to how you work with framework integrations. There are more details on each of these changes in later sections.
+
+- In Apollo Server 3, you pass your `context` function to the `ApolloServer` constructor. The parameters received by this function depend on which `ApolloServer` subclass you are using. Because Apollo Server 4 does not use subclassing, you no longer pass your `context` function to the `ApolloServer` constructor; instead, you pass it to your integration function (eg, `expressMiddleware` or `startStandaloneServer`).
+- In Apollo Server 3, the framework integrations automatically set up their framework's HTTP body parsing and CORS header writing for you, and you could configure that functionality in a framework-specific way via the Apollo Server API. In Apollo Server 4, it's your responsibility to set up these standard features yourself. Specifically, when using `expressMiddleware`, you should install the `body-parser` and `cors` npm packages and use them in your Express app, just like with any other JSON-based API server. Note that `startStandaloneServer` does set up these packages for you automatically, but does not allow you to configure their behavior.
+- In Apollo Server 3, some framework integrations take a `path` option that lets you configure on what URL path they listen on, with a default of `/graphql`. This feature overlapped with the router features defined by web frameworks and has been removed in Apollo Server 4. This means that to provide the same API when migrating, you'll want to explicitly mount `expressMiddleware` at `/graphql`. (Note that the AS3 batteries-included server `apollo-server` listened on all paths, not just `/graphql`.)
+
+The follow sections show how servers using `apollo-server` and `apollo-server-express` need to change to use the new API.
+
+### Migrating from `apollo-server`
+
+If you used the batteries-included `apollo-server` package in Apollo Server 3, you'll want to use the `startStandaloneServer` function in Apollo Server 4. Call `startStandaloneServer(server)` instead of `server.listen()`. Note that you cannot configure CORS settings with `startStandaloneServer`; if you want to change CORS settings from the default, use `expressMiddleware` instead.
+
+<MultiCodeBlock>
```

Comment:
CodeColumns is too narrow, as you suggest. I would rather not put this particular set in <ExpansionPanel> since it's arguably the most important set of examples in the whole doc... ExpansionPanel is nice for reducing scrolling but also makes it harder to find examples of what you're looking for with cmd-F, and I'd rather avoid it in the migration guide for that purpose (which is a place I can image loading up and searching for the particular symbol that is causing an error).

Reply:
Yeah, that's a _really_ good point. I can't think of a better answer at the moment so I'll resolve this :)